src/genesis/mod.rs

```diff
@@ -139,42 +142,65 @@ where
 /// header.
 async fn load_and_retrieve_header<DB, R>(
     store: DB,
-    mut reader: R,
+    reader: R,
     skip_load: bool,
 ) -> anyhow::Result<(Vec<Cid>, Option<usize>)>
 where
     DB: Blockstore + Send + Sync + 'static,
     R: AsyncRead + Send + Unpin,
 {
     let result = if skip_load {
-        (CarReader::new(&mut reader).await?.header.roots, None)
+        (CarReader::new(reader).await?.header.roots, None)
     } else {
-        let (roots, n_records) = forest_load_car(store, &mut reader).await?;
+        let (roots, n_records) = forest_load_car(store, reader).await?;
         (roots, Some(n_records))
     };
 
     Ok(result)
 }
 
+fn car_stream<R: futures::AsyncRead + Send + Unpin>(
+    mut reader: CarReader<R>,
+) -> impl Stream<Item = anyhow::Result<fvm_ipld_car::Block>> {
+    try_stream! {
+        while let Some(block) = reader.next_block().await? {
+            yield block;
+        }
+    }
```

Comment:
this can be a `futures::stream::unfold` instead of using a macro: 
https://github.com/ChainSafe/forest/blob/fa1126a9596eed07fe0aef951b7225daf4626a8e/benches/car-reader.rs#L29-L30

Reply:
Nice.